FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/c_sdk/platform/freertos/transport/src/plaintext_freertos.c

```diff
@@ -31,129 +31,70 @@
+                                        receiveTimeoutMs,
+                                        sendTimeoutMs );
+
+        /* A non zero status is an error. */
         if( socketStatus != 0 )
         {
-            LogError( ( "Failed to establish TCP Connection: ReturnCode=%d.", socketStatus ) );
+            LogError( ( "Failed to connect to %s with error %d.",
+                        pHostName,
+                        socketStatus ) );
+            plaintextStatus = PLAINTEXT_TRANSPORT_CONNECT_FAILURE;
         }
     }
 
-    if( socketStatus == 0 )
-    {
-        /* Set socket receive timeout. */
-        transportTimeout = pdMS_TO_TICKS( receiveTimeoutMs );
-        /* Setting the receive block time cannot fail. */
-        ( void ) FreeRTOS_setsockopt( tcpSocket,
-                                      0,
-                                      FREERTOS_SO_RCVTIMEO,
-                                      &transportTimeout,
-                                      sizeof( TickType_t ) );
-
-        /* Set socket send timeout. */
-        transportTimeout = pdMS_TO_TICKS( sendTimeoutMs );
-        /* Setting the send block time cannot fail. */
-        ( void ) FreeRTOS_setsockopt( tcpSocket,
-                                      0,
-                                      FREERTOS_SO_SNDTIMEO,
-                                      &transportTimeout,
-                                      sizeof( TickType_t ) );
-    }
-
-    /* Clean up on failure. */
-    if( socketStatus != 0 )
-    {
-        if( tcpSocket != FREERTOS_INVALID_SOCKET )
-        {
-            FreeRTOS_closesocket( tcpSocket );
-        }
-    }
-    else
-    {
-        /* Set the socket. */
-        pNetworkContext->tcpSocket = tcpSocket;
-        LogDebug( ( "TCP Connection to %s established.", pHostName ) );
-    }
-
-    return socketStatus;
+    return plaintextStatus;
 }
 
-void Plaintext_FreeRTOS_Disconnect( const NetworkContext_t * pNetworkContext )
+PlaintextTransportStatus_t Plaintext_FreeRTOS_Disconnect( const NetworkContext_t * pNetworkContext )
 {
-    BaseType_t waitForShutdownLoopCount = 0;
-    uint8_t pDummyBuffer[ 2 ];
+    PlaintextTransportStatus_t plaintextStatus = PLAINTEXT_TRANSPORT_SUCCESS;
 
-    if( pNetworkContext->tcpSocket != FREERTOS_INVALID_SOCKET )
+    if( pNetworkContext == NULL )
     {
-        /* Initiate graceful shutdown. */
-        ( void ) FreeRTOS_shutdown( pNetworkContext->tcpSocket, FREERTOS_SHUT_RDWR );
-
-        /* Wait for the socket to disconnect gracefully (indicated by FreeRTOS_recv()
-         * returning a FREERTOS_EINVAL error) before closing the socket. */
-        while( FreeRTOS_recv( pNetworkContext->tcpSocket, pDummyBuffer, sizeof( pDummyBuffer ), 0 ) >= 0 )
+        LogError( ( "pNetworkContext cannot be NULL." ) );
+        plaintextStatus = PLAINTEXT_TRANSPORT_INVALID_PARAMETER;
+    }
+    else
+    {
+        if( pNetworkContext->tcpSocket != FREERTOS_INVALID_SOCKET )
```

Comment:
should  `pNetworkContext->tcpSocket == FREERTOS_INVALID_SOCKET` also result in PLAINTEXT_TRANSPORT_INVALID_PARAMETER?

Reply:
Yes. Good catch. Thanks